### DISCUSSION TITLE
`MaterialGrid` with same material type is not working

### DISCUSSION THREAD

**Discussion by ramji28** [NanoComp/meep#2812] [Q&A]
Dear Meep community,

I was trying to build a metasurface layer with the `MaterialGrid` class. The layer consists of two mediums. I calculated the absorption spectrum of the metasurface built with and without `MaterialGrid`.  If the two mediums which make up the layer are of different material types (`LorentzianSusceptibility` and frequency-independent isotropic ε), then the absorption spectrum with and without the Material grid matches. However, if the two mediums are mediums of the same material type (`LorentzianSusceptibility`), then the absorption spectrum with and without the `MaterialGrid` does not match (see the plots below). Can someone help me with this issue?
Thank you in advance.

![git_no_loaded](https://github.com/NanoComp/meep/assets/87534556/0150c4c5-d465-41df-b4f6-ae2bda7d3fba)
![git_loaded](https://github.com/NanoComp/meep/assets/87534556/43a1fa05-95b3-4c64-b839-723f650fc8ac) 

   

    eps_inf = 1.1
    num_lorentzians = 2
    E_susceptibilities = []
    ps = [37.0880, 0.0000, 0.0483, 0.0000, 8.9469, 5.8681]
    for n in range(num_lorentzians):
        mymaterial_freq = ps[3 * n + 1]
        mymaterial_gamma = ps[3 * n + 2]
        if mymaterial_freq == 0:
            mymaterial_sigma = ps[3 * n + 0]
            E_susceptibilities.append(mp.DrudeSusceptibility(frequency=1.0, gamma=mymaterial_gamma, sigma=mymaterial_sigma))
        else:
            mymaterial_sigma = ps[3 * n + 0] / mymaterial_freq**2
            E_susceptibilities.append(
                mp.LorentzianSusceptibility(
                    frequency=mymaterial_freq,
                    gamma=mymaterial_gamma,
                    sigma=mymaterial_sigma,
                )
            )

    Medium1 = mp.Medium(epsilon=eps_inf, E_susceptibilities=E_susceptibilities)

    Medium2 = mp.Medium(index=1)

    w0 = np.array([1390])  # in cm^-1
    sqrt_s = np.array([185.42])  # in cm^-1
    gamma = np.array([20])  # in cm^-1
    e_inf = 1  # epsilon infinity

    w0_um = 1e4/np.array(w0)  # convert to um
    gamma_um = 1e4/(np.array(gamma))  # convert to um   
    s = np.square(sqrt_s)  # in cm^-2
    sigma = (s/np.square(w0))  # unitless

    f_n = 1/(w0_um)  # convert to frequency
    f_gamma = 1/(gamma_um)  # convert to frequency

    susc = [mp.LorentzianSusceptibility(frequency=f_n[0], gamma=f_gamma[0], sigma=sigma[0]),]
                
    Medium3 = mp.Medium(epsilon=e_inf, E_susceptibilities=susc)`

**Comment by smartalecH** [comment]
Are you passing the dispersive materials via the `extra_materials` parameter in your `Simulation` constructor? Meep can't pull dispersive materials from `MaterialGrids` directly. 

Edit: hmm looks like you are.
